src/main/java/org/amahi/anywhere/activity/AuthenticationActivity.java

```diff
@@ -59,13 +63,47 @@
 {
 	@Inject
 	AmahiClient amahiClient;
-
+  ImageView imageView;
+  
 	@Override
 	protected void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
 		setContentView(R.layout.activity_authentication);
 
-		setUpInjections();
+    getUsernameEdit().setOnTouchListener(new View.OnTouchListener() {
+
+            @Override
+            public boolean onTouch(View view, MotionEvent motionEvent) {
+                getUsernameEdit().setOnFocusChangeListener(new View.OnFocusChangeListener() {
+                    @Override
+                    public void onFocusChange(View view, boolean b) {
+                        if (b) {
+                            imageView.setImageDrawable(null);
```

Comment:
Instead of making the image drawable null you should set the VIsibility property of the ImageView. Set it to View.GONE when you need to hide the IV and to bring it back set it to View.VISIBLE.